src/Symfony/Component/Security/Core/Authentication/Token/UserAuthorizationCheckerToken.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Authentication\Token;
+
+use Symfony\Component\Security\Core\User\UserInterface;
+
+/**
+ * UserAuthorizationCheckerToken implements a token used for checking authorization.
+ *
+ * @author Nate Wiebe <nate@northern.co>
+ */
+class UserAuthorizationCheckerToken extends AbstractToken
```

Comment:
This token implementation should be `@internal` because it's meant to be used by `UserAuthorizationChecker` only. We could also make it `final` for that matter.

Reply:
If we make the token class internal, we should at least have a non-internal marker interface to distinguish special tokens that don't correlate to the current logged in user, to allow voters to distinguish them in case they cannot support that case (the special case done for AuthenticatedVoter permissions might also have an equivalent in a 2FA voter for instance). Otherwise, voters will have to do a `$token instanceof UserAuthorizationCheckerToken` check, which will get warned about if the class is internal (because it would rely on an internal API)

Reply:
Good point.

Reply:
Went with `OfflineTokenInterface`. Let me know if you have a better name.